### PR TITLE
fix(research): recover from section array-cap overflow (Firm Overview / Exec Summary failures)

### PR DESCRIPTION
**Pull request by ewise123**
## Summary
- Fixes research jobs getting stuck on **Firm Overview** and **Executive Summary** in Azure prod (reported for the Ares Management PE report).
- Root cause (confirmed from prod `BugReport` records, all attempts 3/3): the model returns 6+ strategic priorities for a large firm, but `companyOverviewOutputSchema` capped `strategic_priorities.priorities` at `.max(5)`. Zod threw `too_big`, the section failed every retry identically, and because `exec_summary` depends on `company_overview` the failure cascaded — both sections appeared stuck.
- Not environmental: the earlier "Render works" assumption was a red herring (Ares was never run on Render). This is a deterministic, whole-class brittleness bug — ~25 hard `.max()` caps across the section schemas can each do the same.

## Changes
- **`orchestrator-utils.ts`** — new pure `clampArrayOverages(candidate, zodError)` helper: when validation fails *solely* because arrays exceeded their caps, truncate those arrays to the schema's own maximum (using the Zod error's path + `maximum`). Genuine schema errors still surface.
- **`orchestrator.ts`** — `parseStageOutput` now routes both the `financial_snapshot` and generic branches through `validateWithArrayClamp`, which clamps + re-validates once before failing. Logs `[array-clamp]` when it recovers.
- **`prompts/validation.ts`** — raise `strategic_priorities.priorities` cap `5 → 8` (the old 5 was arbitrary; large firms legitimately have more).
- **`prompts/company-overview.ts`** — prompt guidance updated `3-5 → 3-8` to match.
- Synced the committed compiled artifacts `prompts/validation.js` and `prompts/company-overview.js` with their `.ts` sources (vitest resolves the `../../prompts/*.js` imports to these committed files).

## Testing
- [x] Unit tests

Details:
- `npm test` → **336/336 pass**; `tsc --noEmit` clean.
- Added 8 tests in `orchestrator-utils.test.ts`, including a reproduction that drives the **real** `companyOverviewOutputSchema`: >8 priorities fails with the exact prod error (`too_big` at `["strategic_priorities","priorities"]`), and the clamp truncates to 8 and re-validates. Also covers top-level arrays, multi-array clamping, immutability, and the "don't clamp when other errors exist" guard.
- Full end-to-end (live orchestrator run for Ares) needs the prod API key + DB, so verification is at the exact `parseStageOutput` failure point rather than the whole pipeline.

## Changelog
- [x] Updated `CHANGELOG.md` under `[Unreleased]`

## Screenshots / Video (if UI)
- n/a (backend)

## Risk / Rollback
- Risk: **Low.** The clamp only activates on an otherwise-fatal validation failure and only truncates arrays to limits the schema already enforced, so successful paths are unchanged. The cap raise (5→8) only widens what's accepted.
- Note: clamping is silent truncation to the schema max; it logs `[array-clamp]` when it fires so overages are observable.
- Rollback plan: revert this branch's two commits; no migrations or data changes.

## Follow-up (out of scope)
- `backend/prompts/` commits compiled `.js` artifacts alongside `.ts`; vitest resolves imports to the committed `.js`, so tests validate stale compiled output unless the `.js` is regenerated. Prod is unaffected (Docker recompiles `.ts → dist/`). Suggest gitignoring + regenerating them or adding a build step.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of larger array-based inputs so oversized lists are trimmed and rechecked instead of causing validation failures.
  * Increased the allowed number of strategic priorities in company overview content from 5 to 8, reducing truncation for companies with more priorities.
  * Updated validation behavior so valid content is more likely to pass when only list length is the issue.

### CODE REVIEW

File: docs/news-intelligence.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                                       ▼
+                       drop tag-only matches; keep entity hits
+                                       ▼
+              store Article (+ sources) and link to matching users
+                                       ▼
+        ┌──────────────┬───────────────┬──────────────────────────┐
+        ▼              ▼               ▼                          ▼
+   Dashboard     Pin/Dismiss/      Ad-hoc search /         Digest export
+  (per-user)      Archive          Deep dive (per-user)    (PDF/MD/DOCX → "sent")
+```
```
[the remark this answers is not in the document]
Comment (ewise123) [reply docs/news-intelligence.md]: Good flag, but this file was swept into the PR unintentionally by `git add -A` (it was an untracked file predating this work) and is unrelated to the array-cap fix. I've removed `docs/news-intelligence.md` from the branch (kept on disk for a separate PR), so the MD040 nit is moot here.

**Comment by ewise123** [comment]
Addressed CodeRabbit feedback:
- **Edge-case tests** (`clampArrayOverages`): added coverage for deeply nested object paths (`a.b.c`), paths containing an array index (`['items', 0, 'subitems']`), and a shape-mismatch case that exercises the `null` bail-out. Suite now 339/339.
- **`docs/news-intelligence.md` MD040**: that file was included by accident (`git add -A` picked up a pre-existing untracked file) — removed from this PR rather than relabeled.
